aws/resource_aws_ecs_cluster_test.go

```diff
@@ -315,6 +351,69 @@ func testAccCheckAWSEcsClusterExists(resourceName string, cluster *ecs.Cluster)
 	}
 }
 
+func testAccCheckAWSEcsClusterCapacityProviders(cluster *ecs.Cluster, expectedProviderNames []string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).ecsconn
+
+		input := &ecs.DescribeClustersInput{
+			Clusters: []*string{cluster.ClusterArn},
+		}
+
+		resp, err := conn.DescribeClusters(input)
+		if err != nil {
+			return fmt.Errorf("error describing ECS Cluster: %s", err)
+		}
+
+		var actual *ecs.Cluster
+		for _, c := range resp.Clusters {
+			if aws.StringValue(c.ClusterArn) == *cluster.ClusterArn {
+				actual = c
+				break
+			}
+
+		}
+		if actual == nil {
+			return fmt.Errorf("ECS Cluster has disappeared")
+		}
```

Comment:
The `*ecs.Cluster` object passed in should always be fresh from the `testAccCheckAWSEcsClusterExists()` right before so seems extraneous to repeat the lookup (and this is why we like returning the API response objects from `testAccCheck...Exists()` functions 😄 ).

Reply:
I think that was copypasta from somewhere